packages/arb-token-bridge-ui/tests/e2e/specs/importToken.cy.ts

```diff
@@ -0,0 +1,327 @@
+          .click({ scrollBehavior: false })
+
+        cy.findByText('Arbed CMC List').should('be.visible')
+        cy.get('[data-cy="toggle Arbed CMC List"]')
+          .parent()
+          .click({ scrollBehavior: false })
+        cy.get('[data-cy="toggle Arbed CMC List"]').should('be.checked')
+
+        cy.findByRole('button', { name: 'Back to Select Token' })
+          .should('be.visible')
+          .click({ scrollBehavior: false })
+
+        // Select the UNI token
+        cy.findByPlaceholderText(/Search by token name/i)
+          .should('be.visible')
+          .type('UNI', { scrollBehavior: false })
+        cy.findByText('Uniswap').click({ scrollBehavior: false })
+
+        // UNI token should be selected now and popup should be closed after selection
+        cy.findByRole('button', { name: 'Select Token' })
+          .should('be.visible')
+          .should('have.text', 'UNI')
+      })
+    })
+
+    context('Add button is grayed', () => {
+      // log in to metamask
+      before(() => {
+        cy.login({ networkType: 'L1' })
+      })
+
+      after(() => {
+        // after all assertions are executed, logout and reset the account
+        cy.logout()
+      })
+
+      it('should disable Add button if address is too long/short', () => {
+        cy.findByRole('button', { name: 'Select Token' })
+          .should('be.visible')
+          .should('have.text', 'ETH')
+          .click({ scrollBehavior: false })
+
+        // open the Select Token popup
+        cy.findByPlaceholderText(/Search by token name/i)
+          .as('searchInput')
+          .should('be.visible')
+          .type(ERC20TokenAddressL1.slice(0, -1), { scrollBehavior: false })
+
+        // Add button should be disabled
+        cy.findByRole('button', { name: 'Add New Token' })
+          .should('be.visible')
+          .should('be.disabled')
+          .as('addButton')
+
+        // Add last character
+        cy.get('@searchInput').type(
+          `{moveToEnd}${ERC20TokenAddressL1.slice(-1)}`,
+          {
+            scrollBehavior: false
+          }
+        )
+        // Add button should be enabled
+        cy.get('@addButton').should('be.enabled')
+
+        // Add one more character
+        cy.get('@searchInput').type(`{moveToEnd}a`, {
+          scrollBehavior: false
+        })
+        // Add button should be disabled
+        cy.get('@addButton').should('be.disabled')
+      })
+    })
+  })
+
+  context('User import token through URL', () => {
+    afterEach(() => {
+      // after all assertions are executed, logout and reset the account
+      cy.logout()
+      resetSeenTimeStampCache()
+    })
+
+    context('User uses L1 address', () => {
+      it('should import token through its L1 address', () => {
+        cy.login({
+          networkType: 'L1',
+          url: '/',
+          qs: {
+            token: ERC20TokenAddressL1
+          }
+        })
+
+        // waiting for metamask notification to disappear
```

Comment:
MetaMask showing a successful login notification that keep the focus on it?

Tried bunch of options like `cy.switchToCypressWindow` to regain focus but nothing does it.

Feel free to suggest solutions.